scapy/automaton.py

```diff
@@ -828,7 +828,7 @@ def _do_iter(self):
                     raise self.ErrorState("Reached %s: [%r]" % (self.state.state, state_output),  # noqa: E501
                                           result=state_output, state=self.state.state)  # noqa: E501
                 if self.state.final:
-                    raise StopIteration(state_output)
+                    return state_output
```

Comment:
We actually have to change the way we were returning this info, because on Python 3.7, no other thing than the above will work, but Python 2.7 will fail at compiling it (so not even a if/else will work)